### PR TITLE
Added free_symbols method

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: csympy/lib/csympy.pxd [hunk 1]
```diff
@@ -12,6 +12,17 @@ cdef extern from 'gmpxx.h':
     cdef cppclass mpq_class:
         mpq_class()
 
+cdef extern from "<set>" namespace "std":
+    cdef cppclass set[T, U]:
```
Comment (isuruf) [inline csympy/lib/csympy.pxd]: Cython's `set.pxd` file does not support two template arguments to `set`, so it is defined here with minimal methods.

Comment (certik) [inline csympy/lib/csympy.pxd]: Right.

Comment (certik) [inline csympy/lib/csympy.pxd]: I think you should put this into a comment in the file.

**Comment by certik** [comment]
Otherwise +1.

File: csympy/lib/csympy.pxd (hunk 1 above)
Comment (isuruf) [inline csympy/lib/csympy.pxd]: Added the comment

**Comment by certik** [comment]
Thanks!

**Pull request merged**